### ISSUE TITLE
JSON encode error: {bad_term,{packet,raw}}

### ISSUE THREAD

**Issue by veetow** [rabbitmq/rabbitmq-management#363]
Getting these errors when attempting to use the Management plugin on erlang-19.3.0 and rabbitmq-server-3.6.7

```
=ERROR REPORT==== 20-Mar-2017::16:17:01 ===
JSON encode error: {bad_term,{packet,raw}}
While encoding:
[{management_version,<<"3.6.7">>},
 {rates_mode,basic},
 {exchange_types,
     [[{name,<<"headers">>},
       {description,
           <<"AMQP headers exchange, as per the AMQP specification">>},
       {enabled,true}],
      [{name,<<"topic">>},
       {description,<<"AMQP topic exchange, as per the AMQP specification">>},
       {enabled,true}],
      [{name,<<"fanout">>},
       {description,<<"AMQP fanout exchange, as per the AMQP specification">>},
       {enabled,true}],
      [{name,<<"direct">>},
       {description,<<"AMQP direct exchange, as per the AMQP specification">>},
       {enabled,true}]]},
 {rabbitmq_version,<<"3.6.7">>},
 {cluster_name,<<"poe-queue">>},
 {erlang_version,<<"19.3">>},
 {erlang_full_version,
     <<"Erlang/OTP 19 [erts-8.3] [source] [64-bit] [smp:2:2] [async-threads:64] [hipe] [kernel-poll:true]">>},
 {message_stats,
     {struct,
         [{publish,2},
          {publish_details,[{rate,0.0}]},
          {confirm,0},
          {confirm_details,[{rate,0.0}]},
          {return_unroutable,0},
          {return_unroutable_details,[{rate,0.0}]},
          {disk_reads,0},
          {disk_reads_details,[{rate,0.0}]},
          {disk_writes,0},
          {disk_writes_details,[{rate,0.0}]},
          {get,0},
          {get_details,[{rate,0.0}]},
          {get_no_ack,0},
          {get_no_ack_details,[{rate,0.0}]},
          {deliver,2},
          {deliver_details,[{rate,0.0}]},
          {deliver_no_ack,4},
          {deliver_no_ack_details,[{rate,0.0}]},
          {redeliver,0},
          {redeliver_details,[{rate,0.0}]},
          {ack,0},
          {ack_details,[{rate,0.0}]},
          {deliver_get,6},
          {deliver_get_details,[{rate,0.0}]}]}},
 {queue_totals,
     {struct,
         [{messages_ready,0},
          {messages_ready_details,[{rate,0.0}]},
          {messages_unacknowledged,0},
          {messages_unacknowledged_details,[{rate,0.0}]},
          {messages,0},
          {messages_details,[{rate,0.0}]}]}},
 {object_totals,
     {struct,
         [{consumers,20},
          {queues,6},
          {exchanges,10},
          {connections,19},
          {channels,32}]}},
 {statistics_db_event_queue,0},
 {node,'rabbit@bunny01d-ops-08'},
 {listeners,
     [[{node,'rabbit@bunny01d-ops-08'},
       {protocol,amqp},
       {ip_address,<<"::">>},
       {port,5672},
       {socket_opts,
           [binary,
            {packet,raw},
            {reuseaddr,true},
            {backlog,128},
            {nodelay,true},
            {exit_on_close,false},
            {keepalive,false},
            {linger,[true,0]}]}],
      [{node,'rabbit@bunny01d-ops-08'},
       {protocol,'amqp/ssl'},
       {ip_address,<<"::">>},
       {port,5671},
       {socket_opts,
           [binary,
            {packet,raw},
            {reuseaddr,true},
            {backlog,128},
            {nodelay,true},
            {exit_on_close,false},
            {keepalive,false},
            {linger,[true,0]},
            {versions,['tlsv1.2','tlsv1.1']},
            {cacertfile,<<"/etc/pki/tls/certs/PlatformEng-CA.pem">>},
            {certfile,<<"/etc/pki/tls/certs/rabbitmq.pem">>},
            {keyfile,<<"/etc/pki/tls/private/rabbitmq.key">>},
            {verify,verify_peer},
            {fail_if_no_peer_cert,false}]}],
      [{node,'rabbit@bunny01d-ops-08'},
       {protocol,clustering},
       {ip_address,<<"::">>},
       {port,25672},
       {socket_opts,[]}],
      [{node,'rabbit@bunny01d-ops-08'},
       {protocol,https},
       {ip_address,<<"::">>},
       {port,15671},
       {socket_opts,
           [{port,15671},
            {ssl,true},
            {ssl_opts,
                [{cacertfile,"/etc/pki/tls/certs/PlatformEng-CA.pem"},
                 {certfile,"/etc/pki/tls/certs/rabbitmq.pem"},
                 {keyfile,"/etc/pki/tls/private/rabbitmq.key"},
                 {versions,['tlsv1.2','tlsv1.1']}]}]}]]},
 {contexts,
     [[{node,'rabbit@bunny01d-ops-08'},
       {description,<<"RabbitMQ Management">>},
       {path,<<"/">>},
       {port,<<"15671">>},
       {ssl,<<"true">>},
       {ssl_opts,
           [{cacertfile,<<"/etc/pki/tls/certs/PlatformEng-CA.pem">>},
            {certfile,<<"/etc/pki/tls/certs/rabbitmq.pem">>},
            {keyfile,<<"/etc/pki/tls/private/rabbitmq.key">>},
            {versions,<<"['tlsv1.2','tlsv1.1']">>}]}]]}]

```

Not sure if related, but I'm also seeing these in the server logs:

```
=ERROR REPORT==== 20-Mar-2017::16:21:33 ===
** Generic server connection_coarse_metrics_metrics_collector terminating
** Last message in was collect_metrics
** When Server state == {state,connection_coarse_metrics,5000,
                               {[{605,5},{3600,60}],
                                [{605,5}],
                                [{605,5},{3660,60},{29400,600},{86400,1800}]},
                               basic,
                               #Fun<rabbit_mgmt_metrics_collector.0.114916649>,
                               #Fun<rabbit_mgmt_metrics_collector.1.114916649>,
                               {dict,0,16,16,8,80,48,
                                     {[],[],[],[],[],[],[],[],[],[],[],[],[],
                                      [],[],[]},
                                     {{[],[],[],[],[],[],[],[],[],[],[],[],[],
                                       [],[],[]}}}}
** Reason for termination ==
** {badarith,
       [{rabbit_mgmt_metrics_collector,sum_entry,2,
            [{file,"src/rabbit_mgmt_metrics_collector.erl"},{line,550}]},
        {rabbit_mgmt_metrics_collector,'-insert_entry_op/4-fun-0-',2,
            [{file,"src/rabbit_mgmt_metrics_collector.erl"},{line,529}]},
        {dict,update_bkt,4,[{file,"dict.erl"},{line,319}]},
        {dict,on_bucket,3,[{file,"dict.erl"},{line,411}]},
        {dict,update,4,[{file,"dict.erl"},{line,314}]},
        {rabbit_mgmt_metrics_collector,insert_entry_op,4,
            [{file,"src/rabbit_mgmt_metrics_collector.erl"},{line,528}]},
        {lists,foldl,3,[{file,"lists.erl"},{line,1263}]},
        {rabbit_mgmt_metrics_collector,aggregate_entry,4,
            [{file,"src/rabbit_mgmt_metrics_collector.erl"},{line,200}]}]}


=CRASH REPORT==== 20-Mar-2017::16:21:33 ===
  crasher:
    initial call: rabbit_mgmt_metrics_collector:init/1
    pid: <0.1147.0>
    registered_name: connection_coarse_metrics_metrics_collector
    exception exit: {badarith,
                        [{rabbit_mgmt_metrics_collector,sum_entry,2,
                             [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                              {line,550}]},
                         {rabbit_mgmt_metrics_collector,
                             '-insert_entry_op/4-fun-0-',2,
                             [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                              {line,529}]},
                         {dict,update_bkt,4,[{file,"dict.erl"},{line,319}]},
                         {dict,on_bucket,3,[{file,"dict.erl"},{line,411}]},
                         {dict,update,4,[{file,"dict.erl"},{line,314}]},
                         {rabbit_mgmt_metrics_collector,insert_entry_op,4,
                             [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                              {line,528}]},
                         {lists,foldl,3,[{file,"lists.erl"},{line,1263}]},
                         {rabbit_mgmt_metrics_collector,aggregate_entry,4,
                             [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                              {line,200}]}]}
      in function  gen_server:terminate/7 (gen_server.erl, line 812)
    ancestors: [rabbit_mgmt_agent_sup,rabbit_mgmt_agent_sup_sup,<0.351.0>]
    messages: []
    links: [<0.353.0>]
    dictionary: []
    trap_exit: false
    status: running
    heap_size: 2586
    stack_size: 27
    reductions: 2522
  neighbours:


=SUPERVISOR REPORT==== 20-Mar-2017::16:21:33 ===
     Supervisor: {local,rabbit_mgmt_agent_sup}
     Context:    child_terminated
     Reason:     {badarith,
                     [{rabbit_mgmt_metrics_collector,sum_entry,2,
                          [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                           {line,550}]},
                      {rabbit_mgmt_metrics_collector,
                          '-insert_entry_op/4-fun-0-',2,
                          [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                           {line,529}]},
                      {dict,update_bkt,4,[{file,"dict.erl"},{line,319}]},
                      {dict,on_bucket,3,[{file,"dict.erl"},{line,411}]},
                      {dict,update,4,[{file,"dict.erl"},{line,314}]},
                      {rabbit_mgmt_metrics_collector,insert_entry_op,4,
                          [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                           {line,528}]},
                      {lists,foldl,3,[{file,"lists.erl"},{line,1263}]},
                      {rabbit_mgmt_metrics_collector,aggregate_entry,4,
                          [{file,"src/rabbit_mgmt_metrics_collector.erl"},
                           {line,200}]}]}
     Offender:   [{pid,<0.1147.0>},
                  {id,connection_coarse_metrics_metrics_collector},
                  {mfargs,
                      {rabbit_mgmt_metrics_collector,start_link,
                          [connection_coarse_metrics]}},
                  {restart_type,permanent},
                  {shutdown,30000},
                  {child_type,worker}]
```

**Comment by michaelklishin** [comment]
Duplicate of https://github.com/rabbitmq/rabbitmq-management-agent/issues/34, fixed in `3.6.8`.

**Comment by michaelklishin** [comment]
[Relevant rabbitmq-users thread](https://groups.google.com/d/msg/rabbitmq-users/uxhBb19dMvE/TbpLhgFBFAAJ).
